pkg/build/strategies/sti/sti.go

```diff
@@ -448,6 +448,44 @@ func (b *STI) Execute(command string, user string, config *api.Config) error {
 		NetworkMode:     string(config.DockerNetworkMode),
 	}
 
+	// If there are injections specified, override the original assemble script
+	// and wait till all injections are uploaded into the container that runs the
+	// assemble script.
+	if len(config.Injections) > 0 && command == api.Assemble {
+		injectedFiles, err := util.ExpandInjectedFiles(config.Injections)
+		if err != nil {
+			return err
+		}
+		rmScript, err := util.CreateInjectedFilesRemovalScript(injectedFiles)
+		if err != nil {
+			return err
+		}
+		defer os.Remove(rmScript)
+		glog.V(5).Infof("Waiting for injected files to be copied into assemble container...")
+		opts.CommandOverrides = func(cmd string) string {
+			return fmt.Sprintf("while [ ! -f %q ]; do sleep 0.5; done; %s; result=$?; source %s; exit $result",
+				"/tmp/rm-injections", cmd, "/tmp/rm-injections")
```

Comment:
@rhcarvalho because we wait for that file to exists and then we execute it... i can make it a variable, but I'm not big fun of hiding stuff to variables.